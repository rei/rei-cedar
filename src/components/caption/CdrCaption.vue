<script setup>

import { 
  useCssModule
} from 'vue';

const props = defineProps({
  summary: {
    type: String,
  },
  credit: {
    type: String,
  },
});

const baseClass = 'cdr-caption';
const style = useCssModule();
const summaryClass = 'cdr-caption__summary';
const citeClass = 'cdr-caption__cite';

</script>

<template>
  <div :class="style[baseClass]">
    <p
      v-if="summary"
      :class="style[summaryClass]"
    >
      {{ summary }}
    </p>
    <cite
      v-if="credit"
      :class="style[citeClass]"
    >
      {{ credit }}
    </cite>
  </div>
</template>

<style lang="scss" module src="./styles/CdrCaption.module.scss">
</style>
