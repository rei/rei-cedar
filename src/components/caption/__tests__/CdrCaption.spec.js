import { mount } from '../../../../test/vue-jest-style-workaround.js';
import CdrCaption from '../CdrCaption.vue';

describe('CdrCaption', () => {
  it('renders correctly', () => {
    const wrapper = mount(CdrCaption, {
      propsData: {
        summary: 'This is a summary',
        credit: 'This is a credit',
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
