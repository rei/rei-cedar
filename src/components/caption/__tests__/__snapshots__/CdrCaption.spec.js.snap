// Vitest Snapshot v1

exports[`CdrCaption > renders correctly 1`] = `
<div
  class="cdr-caption"
>
  <p
    class="cdr-caption__summary"
  >
    This is a summary
  </p>
  <cite
    class="cdr-caption__cite"
  >
    This is a credit
  </cite>
</div>
`;

exports[`CdrCaption renders correctly 1`] = `
<div
  class="cdr-caption"
>
  <p
    class="cdr-caption__summary"
  >
    This is a summary
  </p>
  <cite
    class="cdr-caption__cite"
  >
    This is a credit
  </cite>
</div>
`;
